fedimintd/src/bin/main.rs

```diff
@@ -72,7 +75,43 @@ async fn main() -> anyhow::Result<()> {
         registry.init();
     }
 
-    let mut task_group = TaskGroup::new();
+    let mut root_task_group = TaskGroup::new();
+    root_task_group.install_kill_handler();
+
+    let task_group = root_task_group.clone();
+    let (main_shutdown_tx, main_shutdown_rx) = tokio::sync::oneshot::channel::<()>();
+
+    root_task_group
+        .spawn_local("main", move |_task_handle| async move {
```

Comment:
IIRC you can get the `TaskGroup` from the `TaskHandle` to avoid clone.

Reply:
I didn't find a way, but not too bad imo, just an `Arc::clone`.